docs/branches.md

```diff
@@ -11,50 +11,69 @@ pass a branch's name to commands that want a revision as argument. For example,
+in the future).
+
+!!! note "Details: how `fetch` pulls branches"
+
+    Let's say you run `jj git fetch --remote origin`. Let's say that `jj` determines
+    that the remote's "main" branch has been moved so that its target is now ahead
+    of the local record in `main@origin`.
+
+    `jj` will then update `main@origin` to the new target. If `main@origin` is
+    **tracked**, `jj` will also apply the change to the local branch `main`. If the
+    local target has also been moved compared to `main@origin` (probably because you
+    ran `jj branch set main`), then the two updates will be merged. If one is ahead
+    of the other, then that target will become the new target. Otherwise, the local
+    branch will become conflicted (see the ["Conflicts" section](#conflicts) below
+    for details).
+
+Most commands don't show the tracked remote branch if it has the same target as
 the local branch. The local branch (without `@<remote name>`) is considered the
 branch's desired target. Consequently, if you want to update a branch on a
 remote, you first update the branch locally and then push the update to the
 remote. If a local branch also exists on some remote but points to a different
-target there, `jj log` will show the branch name with an asterisk suffix
-(e.g. `main*`). That is meant to remind you that you may want to push the branch
-to some remote.
+target there, `jj log` will show the branch name with an asterisk suffix (e.g.
+`main*`). That is meant to remind you that you may want to push the branch to
+some remote.
 
-By default, the default remote branch (e.g. `main@origin`) will be tracked
-automatically. You can use `jj branch track` to track existing remote branches
-individually, or set `git.auto-local-branch = true` configuration to track all
-new remote branches automatically.
+If you want to know the internals of branch tracking, consult the 
+[Design Doc][design].
 
-### What does `git.auto-local-branch = true` actually do?
+### Terminology summary
 
-Jujutsu's fetch operation consist of several steps. First `jj git fetch` fetches
-all Git refs under `refs/remotes/<remote name>`. Then Jujutsu stores these refs
-as remote tracking branches. Finally, if `git.auto-local-branch = true`, Jujutsu
-creates local branches for them. This is similar to Mercurial, which fetches all
-its bookmarks (equivalent to Git branches) by default.
+- A **remote branch** is a branch ref on the remote. `jj` can find out its
+  actual state only when it's actively communicating with the remote. However,
+  `jj` does store the last-seen position of the remote branch; this is the
+  commit `jj show develop@origin` would show. This notion is completely
+  analogous to Git's "remote-tracking branches".
+- A **tracked (remote) branch** is defined above. You can make a remote branch
+  tracked with the [`jj branch track` command](#manually-tracking-a-branch), for
+  example.
+- A **local tracking branch** is the local branch that `jj` tries to keep in
```

Comment:
Perhaps, it should be "tracking local branch" or "tracking (local) branch". "local tracking" sounds like a branch that tracks local.

I still think it's better to avoid defining "tracking local branch" as a single term because "tracking" is just a state, but I don't have good suggestion here.

Reply:
"tracking (local) branch" SGTM, I think.